proposals/readonly-struct-init.md

```diff
@@ -0,0 +1,161 @@
+# readonly object initializers 
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Below
+
+## Summary
+
+This feature will allow for assigning of `readonly` fields and get only auto-implemented properties
+in object initializers when the type being constructed is a `struct` or `ref struct`.
+
+## Motivation
+
+The intent of `readonly` in C# is to prevent the modification of a value after construction has 
+completed. That is once the code calling `new` has a reference to the value there should be no 
+further modification of `readonly` fields. Until the calling code has a reference to the object
+modifications can continue to be made via `this`.
+
+This makes it even easier to define usable `readonly struct` types in C#. The simplest form of
+declaring the fields now creates a fully usable value:
+
+``` csharp
+readonly struct Point { 
+    readonly int X;
+    readonly int Y; 
+}
+
+var p = new Point { X = 42; Y = 13 };
```

Comment:
should this be a comma instead?

Reply:
Indeed. I'm mixing up my F# and C# syntax here.